crates/cargo-contract/src/cmd/verify.rs

```diff
@@ -62,9 +73,100 @@ impl VerifyCommand {
     pub fn run(&self) -> Result<VerificationResult> {
         let manifest_path = ManifestPath::try_from(self.manifest_path.as_ref())?;
         let verbosity: Verbosity = TryFrom::<&VerbosityFlags>::try_from(&self.verbosity)?;
+        if let Some(path) = &self.contract {
+            self.verify_contract(manifest_path, verbosity, path)
+        } else if let Some(path) = &self.wasm {
+            self.verify_wasm(manifest_path, verbosity, path)
+        } else {
+            anyhow::bail!("Either --wasm or --contract must be specified")
+        }
+    }
+
+    /// Verify `.wasm` binary.
+    fn verify_wasm(
+        &self,
+        manifest_path: ManifestPath,
+        verbosity: Verbosity,
+        path: &PathBuf,
+    ) -> Result<VerificationResult> {
+        // 1. Read code hash binary from the path.
+        let file = File::open(path)
+            .context(format!("Failed to open contract binary {}", path.display()))?;
+
+        let mut reader = BufReader::new(file);
+        let mut buffer: [u8; 32] = [0; 32];
+        reader
+            .read_exact(&mut buffer)
+            .context(format!("Failed to read contract binary {}", path.display()))?;
+
+        let reference_code_hash = CodeHash(buffer);
```

Comment:
This is just loading the first 32 bytes of the wasm file, it is not the code hash.

This could result in a false positive if the first 32 bytes of each file are the same.

Could just do `std::fs::read(path)?` to load the whole file.